packages/store/src/services/store/api/api-client.ts

```diff
@@ -109,13 +109,23 @@ export class ApiClient implements ApiClientInterface {
   }
 
   private handleError(error: unknown, operationName: string): OperationError | ValidationError {
-    if (error instanceof OperationError || error instanceof ValidationError) {
-      return error
-    }
     if (error instanceof ClientError) {
       const requestId = this.extractRequestIdFromError(error)
       return new OperationError(operationName, ErrorCodes.GRAPHQL_API_ERROR, {}, requestId)
     }
+
+    if (error?.constructor?.name === 'GraphQLClientError') {
+      const requestId = this.extractRequestIdFromErrorMessage(error)
+      const errors = (error as {errors?: {extensions?: {code?: string}}[]})?.errors
+      const isUnauthorized = errors?.some((err) => err.extensions?.code === 'UNAUTHORIZED')
+
+      if (isUnauthorized) {
+        return new OperationError(operationName, ErrorCodes.UNAUTHORIZED, {}, requestId)
```

Comment:
It took me a while to figure out you raise the error here without any parameters and then handle and re-raising it later. As result access error for `getStoreDetails` will not be enriched with store details.

I would like it much better if you pass down `params` to this method from the API client. I think that way code would be much more straightforward.

Reply:
oh, for some reason I was thinking `store-copy` would call the export and import apis too, whoops!